src/agent.ts

```diff
@@ -0,0 +1,141 @@
+import { MongoClient } from 'mongodb';
+
+import { BenchmarkRunner } from './BenchmarkRunner';
+import { Client } from './client/Client';
+import { config } from './config';
+import { rand } from './lib/rand';
+import { getAdminUser, getClients } from './macros/getClients';
+import populate from './populate';
+
+export default (): void => {
+	let agents: Client[];
+
+	const b = new (class extends BenchmarkRunner {
+		private skippedPopulate = false;
+
+		private adminUser: Client | undefined;
+
+		async populate() {
+			// Had to duplicate here the getClients() function, because the users from profile() are not ready at this point
+			if (!config.DATABASE_URL) {
+				console.log('Skip populate, no DATABASE_URL');
+				this.skippedPopulate = true;
+				return;
+			}
+
+			console.log('Start populate DB');
+
+			const client = new MongoClient(config.DATABASE_URL);
+
+			try {
+				await client.connect();
+				const db = client.db(config.DATABASE_NAME);
+
+				const users = db.collection('users');
+
+				console.log('Checking if the hash already exists');
+
+				if (await users.findOne({ _id: new RegExp(config.hash) })) {
+					console.log('Task skipped');
+					this.skippedPopulate = true;
+					return;
+				}
+
+				const results = await populate();
```

Comment:
should the populate action also create departments and any other omnichannel related record?